### PR TITLE
Nooba bucket Logging configuration

### PR DESCRIPTION
**Pull request by aspandey**
Changes :
1. We are trying to come up with the s3 based logging in nooba so that we can use the log based replication.
This patch is a work in progress patch towards this. which set the configuration of logging on nooba bucket.

### CODE REVIEW

File: src/server/system_services/schemas/bucket_schema.js [hunk 1]
```diff
@@ -269,6 +269,22 @@ module.exports = {
...
                 writes: { type: 'integer' },
             }
-        }
+        },
+        // Do we need enable/disable option? if yes, how to do that?
+        logging: {
+            type: 'object',
+            required: ['source_bucket', 'log_bucket', 'log_prefix'],
+            properties: {
+                source_bucket: {
```
Comment (dannyzaken) [inline src/server/system_services/schemas/bucket_schema.js:278]: I think that `source_bucket` is redundant. This is a schema for an individual bucket, which is the source bucket

File: src/server/system_services/schemas/bucket_schema.js [hunk 2]
```diff
@@ -269,6 +269,22 @@ module.exports = {
                 reads: { type: 'integer' },
                 writes: { type: 'integer' },
             }
-        }
+        },
+        // Do we need enable/disable option? if yes, how to do that?
```
Comment (dannyzaken) [inline src/server/system_services/schemas/bucket_schema.js:273]: according to [AWS docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketLogging.html) you disable logging by sending an empty logging configuration, so I think it's equal to just deleting an existing one. so in my opinion there is no need to have enable\disable in this case.

in general when we do need it we can just add a boolean flag `enabled` or something similar

**Comment by dannyzaken** [comment]
@aspandey notice that you have lint errors: 
https://github.com/noobaa/noobaa-core/actions/runs/3779960506/jobs/6425638866#step:3:2259

you can run `eslint` locally:
* from `noobaa-core` dir first run `npm install` to install all packages including eslint
* run `npm run lint` to see any lint errors

File: src/api/bucket_api.js [hunk 3]
```diff
@@ -781,6 +781,74 @@ module.exports = {
...
+                        $ref: 'common_api#/definitions/bucket_name'
+                    },
+                    log_bucket: {
+                        $ref: 'common_api#/definitions/bucket_name'
+                    },
+                    log_prefix: {
+                        type: 'string',
+                    },
+                },
+                // auth: {
```
Comment (dannyzaken) [inline src/api/bucket_api.js:801]: why is the auth commented out? this property specifies which roles are allowed to perform the operation, by removing it the operation can be performed unauthenticated

File: src/server/system_services/schemas/bucket_schema.js (hunk 2 above)
Comment (dannyzaken) [reply src/server/system_services/schemas/bucket_schema.js:273]: you can remove the comment

**Review by dannyzaken — COMMENTED** [review]
@aspandey other than the few small comments given, it looks good. 
you also need to sign the commit to fix the failing DCO

**Comment by aspandey** [comment]
Checking it on my local machine and will resend it. Somehow it was not giving any error..

File: src/server/system_services/schemas/bucket_schema.js (hunk 2 above)
Comment (dannyzaken) [reply src/server/system_services/schemas/bucket_schema.js:273]: @aspandey, remove the comment before merging

**Review by dannyzaken — APPROVED** [review]

**Pull request merged**